src/lava/lib/optimization/utils/generators/mis.py

```diff
@@ -16,48 +16,102 @@ class MISProblem:
     convert them to a QUBO formulation.
     """
 
-    def __init__(self, num_vertices: int, connection_prob: float,
-                 seed: int = None):
+    def __init__(self, adjacency_matrix: np.ndarray):
         """
-        Instantiate a new random MIS problem, with the given number of
-        vertices and connection probability.
+        Instantiate a new MIS problem, given the adjacency matrix of a graph.
 
         Parameters
         ----------
-        num_vertices: int
-            Number of vertices of the random graph
-        connection_prob: float
-            Connection probability
-        seed: int, optional
-            Seed for numpy random generator
+        adjacency_matrix:
+            Adjacency matrix of a graph.
         """
-        self._num_vertices = num_vertices
-        self._connection_prob = connection_prob
-        self._seed = seed
-        self._adjacency = self._get_random_graph_mtx()
+        if adjacency_matrix.shape[0] != adjacency_matrix.shape[1]:
+            raise ValueError("The adjacency matrix has to be a square matrix.")
+        self._adjacency = adjacency_matrix
+
+    @property
+    def adjacency_matrix(self) -> np.ndarray:
+        """Returns the adjacency matrix of the graph."""
+        return self._adjacency
 
     @property
     def num_vertices(self) -> int:
         """Returns the number of vertices in the graph."""
-        return self._num_vertices
+        return self._adjacency.shape[0]
 
     @property
-    def connection_prob(self) -> float:
-        """Returns the connection probability of the graph."""
-        return self._connection_prob
+    def num_edges(self) -> int:
+        """Returns the number of edges in the graph."""
+        return np.count_nonzero(self._adjacency)
+
+    @classmethod
+    def from_random_uniform(
+        cls, n: int, m: float, seed: int = 0
+    ) -> "MISProblem":
+        """
+        Instantiate a new MIS problem, based on a random graph.

```

Comment:
can we be explicit about the n and m names? Is better to have the unambiguous names, i.e., num_nodes, num_edges

Reply:
Also, maybe say  "based on a random graph's parameters" instead of "based on a random graph"? the user is passing n and m ;)